packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -958,6 +958,10 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
 
     if (_start != null) {
       _acceptDrag(_start!);
+      if (currentUp != null) {
```

Comment:
It is possible that pointer up before accepting the gesture.
In this case, if the drag started, we should end dragging.

Reply:
This makes sense, though I think instead of explicitly calling `_checkDragEnd` and setting the `_dragState` maybe we can just call `_giveUpPointer()`.

Reply:
We already call `_acceptDrag` and trigger a drag-start and drag-update event, so I think we should call `_checkDragEnd` to trigger a drag-end that matches the previous events.
Also, we should call `_resetTaps` to prevent triggering the tap event at the same time.

Reply:
Understood, then maybe we should also do `_stopDeadlineTimer` and `_pastSlopTolerance = false`.

Reply:
> Understood, then maybe we should also do `_stopDeadlineTimer` and `_pastSlopTolerance = false`.

`_stopDeadlineTimer` was called at line 947.

`_pastSlopTolerance` is false in this case, am I miss something?

Reply:
A little doubt about [this](https://github.com/flutter/flutter/commit/aa2d73b6ea216e5523f784e7f43dc49b94198d21#r105292224).

Reply:
Following up on https://github.com/flutter/flutter/commit/aa2d73b6ea216e5523f784e7f43dc49b94198d21#r105292224

I believe that calculation you noted is correct. My initial use of `_pastSlopTolerance` was mostly to disqualify the recognizer from registering the gesture as a `tap`, and I was looking at `TapGestureRecognizer - PrimaryPointerGestureRecognizer` for this and they both hardcode the slop tolerance to `kTouchSlop` (maybe this also needs to be changed). 

```dart
/// Determine the appropriate hit slop pixels based on the [kind] of pointer.
double computeHitSlop(PointerDeviceKind kind, DeviceGestureSettings? settings) {
  switch (kind) {
    case PointerDeviceKind.mouse:
      return kPrecisePointerHitSlop; //1.0
    case PointerDeviceKind.stylus:
    case PointerDeviceKind.invertedStylus:
    case PointerDeviceKind.unknown:
    case PointerDeviceKind.touch:
    case PointerDeviceKind.trackpad:
      return settings?.touchSlop ?? kTouchSlop; //18.0
  }
}
```

I think changing the implementation to use `computeHitSlop` would change the behavior on `PointerDeviceKind.mouse`, making a drag more likely since the threshold is less. Seems like a reasonable change to me with a precise pointer device.

Reply:
> > Understood, then maybe we should also do `_stopDeadlineTimer` and `_pastSlopTolerance = false`.
> 
> `_stopDeadlineTimer` was called at line 947.
> 
> `_pastSlopTolerance` is false in this case, am I miss something?

I completely overlooked line 947, you are correct about `_stopDeadlineTimer`. I think `_pastSlopTolerance` might be true (since this particular bug is related to drag) if we did not hit `didStopTrackingLastPointer` first which sets it to false. I ran your test case and it seems like we do.

Reply:
I would like to get this PR in but I wonder if there is a different underlying issue. Do you have any thoughts on https://github.com/flutter/flutter/issues/122519#issuecomment-1478514552